### PR TITLE
Support spaces in file paths

### LINKED ISSUE #24: Build dependencies with whitespace in their path are not properly handled
Build dependencies are currently being parsed using the following line:

https://github.com/dflemstr/rust-native-wasm-loader/blob/03a6e58fb23ca3e64904b647225e4b04af1ccb83/src/cargo.js#L77

This will fail if a space exists anywhere in a dependency path. For example, if you have a dependency file that contains something like this:

```text
C:\Users\User\ Name\Documents\project\target\wasm32-unknown-unknown\debug\rust.wasm: C:\Users\User\ Name\Documents\project\src\mod.rs C:\Users\User\ Name\Documents\project\src\lib.rs
```

You wind up trying to add each of the following lines as a dependency:

```text
C:\Users\User
Name\Documents\project\src\mod.rs
C:\Users\User
Name\Documents\project\src\lib.rs
```

Fixing this is a little tricky, since JavaScript regex doesn't support negative lookbehind. My current solution is to reverse the string, and use a negative lookahead instead:

```javascript
const reverseString = str => str.split('').reverse().join('');

const parseDependencies = str =>
    reverseString(str.split(/:\s+/)[1].trim())
        .split(/(?:\s(?!\\))+/)
        .map(reverseString);
```

which correctly parses the above example into

```text
C:\Users\User Name\Documents\project\src\mod.rs
C:\Users\User Name\Documents\project\src\lib.rs
```

Is that sufficient? Or am I missing a better solution?

### PR DESCRIPTION
**Pull request by grind086**
Fixes #24

### CODE REVIEW

File: src/cargo.js [hunk 1]
```diff
@@ -72,12 +84,10 @@ export const handleCargo = async function (self, result) {
 
   const depFile = wasmFile.slice(0, -'.wasm'.length) + '.d';
   const depContents = await fse.readFile(depFile, 'utf-8');
-  for (let line of depContents.split('\n')) {
-    if (line.startsWith(wasmFile) || (jsFile && line.startsWith(jsFile))) {
-      for (let dep of line.split(/:\s+/)[1].split(/\s+/)) {
-        self.addDependency(dep);
-      }
-    }
+  const deps = parseDependencies(depContents, wasmFile.replace(/ /g, '\\ '));
```
Comment (dflemstr) [inline src/cargo.js]: Can the `wasmFile.replace(/ /g, '\\ ')` bit be done inside of `parseDependencies`?  To me it seems like an implementation detail of the dependency file format.

File: src/cargo.js [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
 import fse from 'fs-extra';
 import path from 'path';
 
+export const parseDependencies = (data, baseFile) =>
+  data
+    .split('\n')
+    .filter(str => str.startsWith(baseFile))
+    .map(str => str.slice(str.indexOf(': ') + 2))
+    .map(reverseString)
+    .map(str => str.split(/(?: (?!\\))+/))
```
Comment (dflemstr) [inline src/cargo.js]: Can we avoid reversing each string by using negative lookbehind instead of lookahead here?

File: src/cargo.js (hunk 1 above)
Comment (grind086) [reply src/cargo.js]: Sure thing. That definitely makes more sense.

File: src/cargo.js (hunk 2 above)
Comment (dflemstr) [reply src/cargo.js]: OK makes sense with your current approach then! :+1:

**Pull request merged**